codice/ddf
# DDF-4610 Add checkboxes to result-item and table view to simplify multi-select

#### What does this PR do?
Checkboxes should be added to result-item and table view to give users a clear indicator of multi selection and provide a more user friendly method for multi select. Select/deselect all checkboxes should also be provided. result-item gallery-view will also modified to decrease the thumbnail size and add a label for Result Preview Metacard Attributes
Edit: result-item gallery view styling changes removed
#### Who is reviewing it? 
@Bdthomson @hayleynorton @lamhuy 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@bdeining
@djblue

#### How should this be tested?
- build / run / ingest some test data
- run * search and verify the checkboxes on results and table allow for selecting multiple results
- verify select/deselect all checkboxes function properly (in the result pane and in the header of the table)
- verify no issues with styling in list and gallery view of the results
- see demo video
#### Any background context you want to provide?
#### What are the relevant tickets?

For GH Issues:
Fixes: #4610 

#### Screenshots
[multiselect-checkbox.mp4.zip (demo video)
](https://github.com/codice/ddf/files/3024701/multiselect-checkbox.mp4.zip)

demo preview (first ~8 seconds)
![multiselect-checkbox](https://user-images.githubusercontent.com/39923178/55265383-ac608480-523d-11e9-8362-4a6b6568e89a.gif)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
